.github/workflows/ci.yml

```diff
@@ -139,3 +172,258 @@ jobs:
+
+      - name: Update comment
+        if: ${{ steps.fc.outputs.comment-id != 0 }}
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          comment-id: ${{ steps.fc.outputs.comment-id }}
+          edit-mode: replace
+          body: ${{ steps.get-comment-body.outputs.body }}
+
+  display-pr:
+    needs: compare_mips
+    runs-on: ubuntu-latest
+    if: ${{github.event_name == 'pull_request'}}
+
+    steps:
+      - name: Get comment text
+        uses: actions/download-artifact@v2
+        with:
+          name: issue.md
+
+      - name: Load comment body
+        id: get-comment-body
+        run: |
+          body=$(cat issue.md)
+          body="${body//'%'/'%25'}"
+          body="${body//$'\n'/'%0A'}"
+          body="${body//$'\r'/'%0D'}"
+          echo ::set-output name=body::$body
+      - name: Find Comment
+        uses: peter-evans/find-comment@v1
+        id: fc
+        with:
+          issue-number: ${{github.event.number}}
+          body-includes: This comment was created automatically, please do not change!
+
+      - name: Create comment
+        if: ${{ steps.fc.outputs.comment-id == 0 }}
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          issue-number: ${{github.event.number}}
+          edit-mode: replace
+          body: ${{ steps.get-comment-body.outputs.body }}
+
+      - name: Update comment
+        if: ${{ steps.fc.outputs.comment-id != 0 }}
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          comment-id: ${{ steps.fc.outputs.comment-id }}
+          edit-mode: replace
+          body: ${{ steps.get-comment-body.outputs.body }}
+
+
+
+  display-wiki:
+    needs: compare_mips
+    runs-on: ubuntu-latest
+    if: ${{github.event_name == 'push'}}
+
+    steps:
+      - name: Get comment text
+        uses: actions/download-artifact@v2
+        with:
+          name: wiki.md
+
+      - name: Get performance_metrics.svg
+        uses: actions/download-artifact@v2
+        with:
+          name: performance_metrics.svg
+
+      - name: Update wiki
+        run: |
+          echo "Cloning wiki repo https://github.com/$GITHUB_REPOSITORY.wiki.git"
+          git clone "https://$GITHUB_ACTOR:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY.wiki.git" "wiki"
+          cd wiki
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+          rm -rf ./performance_metrics.svg
+          cp ../wiki.md ./Performance_Metrics.md
+          cp ../performance_metrics.svg ./performance_metrics.svg
+          git add Performance_Metrics.md performance_metrics.svg
+          git commit -m "update perf"
+          git push
+
+
+  delete-artifact:
+    if: always()
+    needs: [display-wiki, display-push, display-pr]
+    runs-on: ubuntu-latest
+    continue-on-error: true
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Delete Artifact
```

Comment:
Till now, I have found two solutions for this. The first one is https://github.com/GeekyEggo/delete-artifact where you need to explicitly mention the name of the artifact you want to delete. Hence we wanted to avoid it.

Regarding this https://github.com/tum-ei-eda/etiss/actions/runs/1240376195, my experiments show me that it succeeds in deleting the artifacts of the past workflow runs, but not the current one. However, that shouldn't be the case. I am still looking into this (as to why this is happening) as well as other methods and will let you know if I get a solution.